tests/search.sh

```diff
@@ -19,7 +19,7 @@ clearCache
 
 ## Search expressions
 
-# Check that empty search string matches all
-nix search -f search.nix '' |grep -q foo
-nix search -f search.nix '' |grep -q bar
-nix search -f search.nix '' |grep -q hello
+# Check that '--all' flag matches all
+nix search -af search.nix |grep -q foo
+nix search --all -f search.nix |grep -q bar
+nix search -af search.nix |grep -q hello
```

Comment:
```suggestion
nix search -af search.nix |grep -q hello
# Check that an empty search string raises an error
nix search -f search.nix || fail "Running 'nix search' with an empty search string should fail"
```

Reply:
Mh I wrote that a bit too fast, should probably rather be something like

```bash
(! nix search -f search.nix ) || fail "Running 'nix search' with an empty search string should fail"
```

Reply:
added and tested, builds successfully